### ISSUE TITLE
cross_validate's signature, location, and purpose

### ISSUE THREAD

**Issue by adrinjalali** [probabl-ai/skore#731]
Right now the signature is 

```py
cross_validate(*args, project, **kwargs)
```

This doesn't play well with auto-complete and IDEs and inspection. As a user it would be much nicer if the signature matches the scikit-learn's signature.

Another thing which would be nice is the import location. The nicest thing as a user would be to have to change only the top level import, as:

```py
from sklearn.model_selection import cross_validate
```

to

```py
from skore.model_selection import cross_validate
```

However, I wonder if it's a good pattern where we'd then need to basically patch a bunch of sklearn methods to add things to them or change their default values.

To me, it would be much more natural to get the output of sklearn's cross_validate, the estimator, and the data, and then generate what we need to show to the user, instead of calculating all of those on the fly in a custom implementation of `cross_validate`.

So in reality, I would remove this implementation, and have something like:

```py
from sklearn.model_selection import cross_validate
cv_results = cross_validate(..., return_estimator=True, return_indices=True)
project.put(X=x, y=y)
project.put("cv_result", cv_results=cv_results)
# We could generate from stored objects (names here), or actual values.
project.generate_scores_from(X="X", y="y", cv_results="cv_results")
```

cc @koaning @glemaitre

**Comment by adrinjalali** [comment]
The docs also don't explain what really this function does: https://probabl-ai.github.io/skore/latest/generated/skore.cross_validate.html

**Comment by adrinjalali** [comment]
What I was suggesting is more or less implemented in https://github.com/probabl-ai/skore/pull/749 now (thanks to @glemaitre )

**Comment by glemaitre** [comment]
So by playing in #749, I got a couple of thoughts.

> save the predictions and/or decision function so that unplanned metrics can be later re-computed

Currently, as far I see, this is not the case. `skore.cross_validate` somehow use some caching because `sklearn.model_selection.cross_validate` will use `_MultimetricScorer` with multi-metric. However, the predictions are not persisted in a `skore` object.

So here, I would say that the wrapping is not necessary because we can manage the cache in a skore object that consume the `cv_results` from scikit-learn. The only necessary thing is that the user needs to be sure to call `sklearn.model_selection.cross_validate` with `return_estimators` and `return_indices` such that we can compute the any new metric on the fly (and persist whatever is required).

> check how it was called to generate warnings

I would say it depends on the warning.

My opinion here is that a function `cross_validate` that raises several time a warning (e.g. imbalanced classes, missing values, etc.) will be discarded by our user. They already ignore those in scikit-learn even when we raise `ConvergenceWarning`. Here, I think what would work better is to make the user pull the information explicitly and so my proposal of:

```python
reporter.linting.data_checks()
reporter.linting.split_checks()
reporter.linting.estimator_checks()
reporter.linting.report_warnings()
...
```

So my question would be "is there any type of warnings that you want to raise at runtime of the cross-validation that you cannot raise in introspection afterwards?"

> capture input data and do some more checks

This might a point where I'm undecided. By not modifying `sklearn.model_selection.cross_validate`, I would request the user to provide me the `cv_results` and the data that was provide at the same time. When this is internal, you avoid a user mistake.

However, I think that a user does not need to advance to provide the right data and target with the following call:

```
cv_results = sklearn.model_selection.cross_validate(estimator, X, y)
reporter = CrossValidationReporter(cv_results, X, y)
reporter.help()
```

**Comment by glemaitre** [comment]
When it comes to activity feed and the skore-ui, I did not put enough thoughts to have a relevant conversation at this stage.

**Comment by GaelVaroquaux** [comment]
@tuscland says:
> skore.cross_validate has no side-effect other than in the activity feed

I think that everything in our activity feed should be stored and searchable.

@glemaitre says:
> a function cross_validate that raises several time a warning (e.g. imbalanced classes, missing values, etc.) will be discarded by our user.

I agree
